src/main/java/com/tecnalia/datausage/service/ContractAgreementService.java

```diff
@@ -222,4 +211,14 @@ public ResponseEntity<List<ContractStore>> getAllOdrlPolicyPersistence() {
 
 	}
 
+	private ContractStore setProfile(String jsonString, ContractStore contractStore) {
```

Comment:
If you provide CA as in our examples folder, which doesn't have a profile, the value is set to null, which causes the null pointer exception when enforcing policies, to be specific in getSecurityProfile method.

Either to set the profile attribute as optional, or change the logic in enforcement.